### DISCUSSION TITLE
[Question]: How to add custom RAG?

### DISCUSSION THREAD

**Discussion by jayaprakash1207** [danny-avila/LibreChat#3494] [Q&A]
### What is your question?

I have a separate service with a post API consisting of the endpoint - /query which takes the text from the input of the user and gives a LLM generated response. I want to integrate this as a custom endpoint. How do I do it?

### More Details

Like I have tried to configure the custom endpoint in the librechat.yml and made some custom changes in the service controllers and app client but I'm facing serveral issues regarding the conversationID and facing few mishaps in the request and response cycle.

It would be great to know a solution for this real quick!

Thanks!

### What is the main subject of your question?

Endpoints

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
I mean, if it works the same as the current RAG server, you would just need to change `RAG_API_URL`

>  I want to integrate this as a custom endpoint. How do I do it?

There's a similar discussion on this topic:

https://github.com/danny-avila/LibreChat/discussions/2839

**Reply by keerthana-5170** [reply]
Hi @danny-avila , working on integrating a similar custom RAG API with Librechat, its working is not similar to current RAG server since it stores embeddings and answers for the pdfs uploaded, whereas our POST RAG API which answers from the pre ingested set of documents from a vector db(which doesn't require any documents to be uploaded from user end). Attaching a working of our RAG architecture.
![Screenshot 2024-08-01 113148](https://github.com/user-attachments/assets/c9a99a3e-9f47-44e6-83b0-ceac2be5f301)
Can you confirm , if such an custom post RAG API can be integrated as an endpoint with libre chat?

**Comment by acbsn** [comment]
this would be extremely useful since I also have a similar set up where my RAG system is already up and running so I just need the answer (and maybe it would be great to be able to display in the UI the Document excerpts used to generate the answer)
